### ISSUE TITLE
Combobox role now works with `button` element

### ISSUE THREAD

**Issue by joaopslins** [mdn/content#36436]
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/combobox_role

### What specific section or headline is this issue about?

Initial paragraph, but it might be fixed elsewhere in the document

### What information was incorrect, unhelpful, or incomplete?

The initial paragraph states:

> The `combobox` role identifies an element as an `input` that controls another element, such as a `listbox` or `grid`, that can dynamically pop up to help the user set the value of that `input`.

Additionally, all the references in the document about the element that is identified by the `combobox` role says that it should be an `input`.

### What did you expect to see?

It seems like the ARIA spec has changed in order to accept `<button role="combobox" />`. So I'd expect to see the information that I can indeed use `button` elements for `role="combobox"`.

### Do you have any supporting links, references, or citations?

See https://github.com/w3c/html-aria/pull/396 for more information.
